GUI/src/gui_qt.py

```diff
@@ -0,0 +1,458 @@
+        self.carma_sim_logs_button.clicked.connect(lambda: self.open_logs("/opt/carma-simulation/logs"))
+        self.toolbar.addWidget(self.carma_sim_logs_button)
+        
+        # Config dropdown
+        self.config_label = QLabel("Select Config:")
+        self.main_layout.addWidget(self.config_label)
+        self.config_combo = QComboBox()
+        self.config_combo.setEnabled(False)
+        self.config_combo.currentTextChanged.connect(self.on_config_selected)
+        self.main_layout.addWidget(self.config_combo)
+        
+        # Horizontal layout for Start and Stop buttons
+        self.button_layout = QHBoxLayout()
+        self.start_button = QPushButton("Start")
+        self.start_button.setEnabled(False)
+        self.start_button.clicked.connect(self.start_simulation)
+        self.button_layout.addWidget(self.start_button)
+        
+        self.stop_button = QPushButton("Stop")
+        self.stop_button.setEnabled(False)
+        self.stop_button.clicked.connect(self.stop_simulation)
+        self.button_layout.addWidget(self.stop_button)
+        
+        self.main_layout.addLayout(self.button_layout)
+        
+        # Log text area
+        self.log_text = QTextEdit()
+        self.log_text.setReadOnly(True)
+        self.main_layout.addWidget(self.log_text)
+        
+        # Styling for colored log messages
+        self.log_text.setStyleSheet("QTextEdit { color: black; }")
+        
+        # Debug environment
+        self.log_message(f"DISPLAY: {os.environ.get('DISPLAY', 'Not set')}")
+        self.log_message(f"QT_QPA_PLATFORM: {os.environ.get('QT_QPA_PLATFORM', 'Not set')}")
+        
+        # Start log queue processor
+        self.process_log_queue()
+
+    def log_message(self, message, tag=None):
+        color = "black"
+        if tag == "error":
+            color = "red"
+        elif tag == "warning":
+            color = "orange"
+        self.log_text.append(f'<span style="color:{color}">{message}</span>')
+        self.log_text.verticalScrollBar().setValue(self.log_text.verticalScrollBar().maximum())
+        print(f"[GUI Log] {message}")  # Print to terminal for debugging
+
+    def process_log_queue(self):
+        try:
+            while not self.log_queue.empty():
+                message, tag = self.log_queue.get_nowait()
+                self.log_message(message, tag)
+        except:
+            pass
+        self.log_text.repaint()  # Ensure GUI updates
+        self.after(100, self.process_log_queue)  # Check queue periodically
+
+    def after(self, ms, func):
+        from PySide6.QtCore import QTimer
+        QTimer.singleShot(ms, func)
+
+    def open_logs(self, path):
+        try:
+            if not os.path.exists(path):
+                self.log_message(f"Error: Log directory {path} does not exist.", "error")
+                return
+            self.log_message(f"Opening log directory: {path}")
+            subprocess.run(["xdg-open", path], check=True)
```

Comment:
Does this block the main UI thread while the file browser is open? Not a huge deal, especially for this initial MVP build, but could be inconvenient in the final implementation if so.

Reply:
It does block but only from the moment the `Open CARMA Logs` or `Open CDASim Logs` button is clicked until the file browser window pops up.